### PR TITLE
Make Travis test the book

### PR DESCRIPTION
**Pull request by Moxinilian**
This PR makes Travis build all the examples in the book as part of the typical testing process.
This is very fast and will make us avoid the documentation nonsense we had on 0.8 release.

### CODE REVIEW

**Comment by LucioFranco** [comment]
Looks like all tests are failing properly here, so should be a fix in the book and good to go. Good stuff mox.

**Comment by Moxinilian** [comment]
Are you kidding me

**Comment by Moxinilian** [comment]
Okay good the git weird stuff is fixed.
I rebased on master to have the book fixed, and added tests to the prefab chapter.
It should now pass the tests properly and be mergeable.

**Comment by Moxinilian** [comment]
As I understood what went wrong last time when writing tests for derive macros in the book, I finally managed to add tests to the `#[derive(SystemData)]` part of the book!

**Comment by jojolepro** [comment]
Travis still failing

**Comment by Moxinilian** [comment]
Yes. However it seems to be caused by a benchmark-as-a-test thing, which is unreliable enough that it might just be a coincidence. I will restart Travis when needed.

File: book/src/prefab.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Prefabs
 
-A `Prefab` in `Amethyst` is at the core a simple list of future entities, where each entry in
```
Comment (torkleyy) [inline book/src/prefab.md]: Finally, thank you! Either Amethyst or `amethyst`, not `Amethyst`. Biggest issue in the book resolved xD

Comment (torkleyy) [reply book/src/prefab.md]: We should test for this on Travis and make the build fail if somebody wrote `Amethyst` somewhere.

**Comment by Moxinilian** [comment]
What is wrong with Travis? It fails linking on all linux builds for no reason at all.
Does anyone here know why?

**Comment by Moxinilian** [comment]
Thank you torkleyy for fixing Travis!
bors r+

**Pull request merged**